File: problem_50a.py
# Codeforces Problem 50A - Domino piling
# Solution: The maximum number of dominoes that can fit on an M x N board
# is simply (M * N) // 2, since each domino covers 2 squares

# Read input
m, n = map(int, input().split())

# Calculate maximum number of dominoes
max_dominoes = (m * n) // 2

# Output result
print(max_dominoes)

File: problem_50a.rs
// Codeforces Problem 50A - Domino piling
// Row-centric solution: M // 2 * N + N // 2 (if M is odd)

use std::io;

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();

    let nums: Vec<i32> = input
        .trim()
        .split_whitespace()
        .map(|s| s.parse().unwrap())
        .collect();

    let m = nums[0];
    let n = nums[1];

    // Row-centric formula:
    // - M // 2 * N: Complete pairs of rows filled with horizontal dominoes
    // - N // 2: If M is odd, fill the remaining row with vertical dominoes
    let mut result = (m / 2) * n;
    if m % 2 == 1 {
        result += n / 2;
    }

    println!("{}", result);
}
